addons/adf/src/main/java/org/apache/falcon/adfservice/ADFJob.java

```diff
@@ -0,0 +1,563 @@
+    public static final String PROCESS_SCRIPTS_PATH = TEMPLATE_PATH_PREFIX
+            + Path.SEPARATOR + "generatedscripts";
+    private static final String DEFAULT_FREQUENCY = "days(1)";
+
+    // Constants for process properties
+    private static final String PROPERTY_START = "<property ";
+    private static final String PROPERTY_NAME = "name=";
+    private static final String PROPERTY_VALUE = "value=";
+    private static final String PROPERTY_END = "/>";
+    private static final String PROPERTY_DOUBLE_QUOTES = "\"";
+
+    public static boolean isADFJobEntity(String entityName) {
+        return entityName.startsWith(ADF_JOB_ENTITY_NAME_PREFIX);
+    }
+
+    public static String getSessionID(String entityName) throws FalconException {
+        if (!isADFJobEntity(entityName)) {
+            throw new FalconException("The entity, " + entityName + ", is not an ADF Job Entity.");
+        }
+        return entityName.substring(ADF_JOB_ENTITY_NAME_PREFIX_LENGTH);
+    }
+
+    /**
+     * Enum for job type.
+     */
+    public static enum JobType {
+        HIVE, PIG, REPLICATION
+    }
+
+    private static enum RequestType {
+        HADOOPMIRROR, HADOOPHIVE, HADOOPPIG
+    }
+
+    public static JobType getJobType(String msg) throws FalconException {
+        try {
+            JSONObject obj = new JSONObject(msg);
+            JSONObject activity = obj.getJSONObject(ADFJsonConstants.ADF_REQUEST_ACTIVITY);
+            if (activity == null) {
+                throw new FalconException("JSON object " + ADFJsonConstants.ADF_REQUEST_ACTIVITY + " not found in ADF"
+                        + " request.");
+            }
+
+            JSONObject activityProperties = activity.getJSONObject(ADFJsonConstants.ADF_REQUEST_TRANSFORMATION);
+            if (activityProperties == null) {
+                throw new FalconException("JSON object " + ADFJsonConstants.ADF_REQUEST_TRANSFORMATION + " not found "
+                        + "in ADF request.");
+            }
+
+            String type = activityProperties.getString(ADFJsonConstants.ADF_REQUEST_TYPE);
+            if (StringUtils.isBlank(type)) {
+                throw new FalconException(ADFJsonConstants.ADF_REQUEST_TYPE + " not found in ADF request msg");
+            }
+
+            switch (RequestType.valueOf(type.toUpperCase())) {
+            case HADOOPMIRROR:
+                return JobType.REPLICATION;
+            case HADOOPHIVE:
+                return JobType.HIVE;
+            case HADOOPPIG:
+                return JobType.PIG;
+            default:
+                throw new FalconException("Unrecognized ADF job type: " + type);
+            }
+        } catch (JSONException e) {
+            throw new FalconException("Error when parsing ADF JSON message: " + msg, e);
+        }
+    }
+
+    public abstract void startJob() throws FalconException;
+    public abstract void cleanup() throws FalconException;
```

Comment:
Is it used only in Replication and rest all the places cleanUpProcess is used? Correct me If I am wrong.